28_FFTBloom/main.cpp

```diff
@@ -1075,8 +1081,20 @@ class FFTBloomApp final : public examples::SimpleWindowedApplication, public app
 		cmdBuf->bindComputePipeline(m_lastAxisFFT_convolution_lastAxisIFFTPipeline.get());
 		// Update padding for run along rows
 		cmdBuf->pushConstants(m_firstAxisFFTPipeline->getLayout(), IShader::E_SHADER_STAGE::ESS_COMPUTE, offsetof(PushConstantData, padding), sizeof(paddingAlongRows), &paddingAlongRows);
-		// One workgroup per row in the lower half of the DFT
-		cmdBuf->dispatch(core::roundUpToPoT(m_marginSrcDim.height) / 2, 1, 1);
+		// One dispatch per channel
+		for (uint32_t channel = 0u; channel < Channels; channel++)
+		{
+			// Update push contants for run
+			uint64_t channelOffsetBytes = uint64_t(channel) * m_imageSecondAxisFFTNumWorkgroups * imageExtent.width * (m_useHalfFloats ? sizeof(complex_t<float16_t>) : sizeof(complex_t<float32_t>));
+
+			pushConstants.channelInfo.currentChannel = channel;
+			pushConstants.channelInfo.channelStartOffsetBytes = channelOffsetBytes;
+
+			cmdBuf->pushConstants(m_firstAxisFFTPipeline->getLayout(), IShader::E_SHADER_STAGE::ESS_COMPUTE, offsetof(PushConstantData, channelInfo), sizeof(pushConstants.channelInfo), &pushConstants.channelInfo);
+
+			// One workgroup per row in the lower half of the DFT
+			cmdBuf->dispatch(core::roundUpToPoT(m_marginSrcDim.height) / 2, 1, 1);
+		}
```

Comment:
but you could have just done a `gl_WorkGroupCount.y==3` dispatch instead!?

and map `channel = gl_WorkGroupID.y` ?

don't push new constants, and compute a new byte offset for the channel, add a `channelStrideBytes` to the push constants instead and compute in the shader.